app/uk/gov/hmrc/cgtpropertydisposals/models/BprRequest.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.models
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+import play.api.libs.json.{Json, OFormat}
+
+import scala.util.{Failure, Success, Try}
+
+final case class BprRequest(nino: String, fname: String, lname: String, dob: String) {
```

Comment:
I think it'd be better to be explicit about the field names so there's no ambiguity, e.g. `forename` instead of `fname`. Also,    date of birth could be better typed as `LocalDate` here which would mean you wouldn't need the `require` - you'd just have a `Format` instance with the correct formatter in it.

Reply:
refactored